### PR TITLE
ActivityLog Alerts and ActionGroups

### PR DESCRIPTION
**Pull request by dkamstra**
Single pull request incorporating feedback from ARM dev team

### CODE REVIEW

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -180,6 +180,8 @@
                   { "$ref": "http://schema.management.azure.com/schemas/2016-04-30-preview/Microsoft.Compute.json#/resourceDefinitions/virtualMachines"},
                   { "$ref": "http://schema.management.azure.com/schemas/2016-04-30-preview/Microsoft.Compute.json#/resourceDefinitions/virtualMachineScaleSets"},
                   { "$ref": "http://schema.management.azure.com/schemas/2016-04-30-preview/Microsoft.Compute.json#/resourceDefinitions/extensions"},
+                  { "$ref": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.ActionGroup.json#/resourceDefinitions/actiongroups" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: should be "actionGroups" with capital "G". same issue in 2015-01-01/deploymentTemplate.json.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 2]
```diff
@@ -180,6 +180,8 @@
...
+                  { "$ref": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json#/resourceDefinitions/activitylogalerts" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: should be "activityLogAlerts".

File: schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  "title": "Microsoft.Insights Activity Log Alerts",
+  "description": "Microsoft Insights Types",
+  "resourceDefinitions": {
+    "activityLogAlerts": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Insights/activitylogalerts"
```
Comment (TianoMS) [inline schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json]: activityLogAlerts.

File: schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  },
+  "definitions": {
+    "ActivityLogAlert": {
+      "type": "object",
+      "properties": {
+        "scopes": {
+          "oneOf": [
+            {
+              "type": "string",
```
Comment (TianoMS) [inline schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json]: Should be "array"?

File: tests/2017-03-01-preview/Microsoft.Insights.ActionGroup.tests.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            }
+          ]
+        }
+      }
+    },
+    {
+      "name": "Action Group Test - Email Only",
+      "definition": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.ActionGroup.json#/resourceDefinitions/actiongroups",
+      "json": {
+        "type": "Microsoft.Insights/activityLogAlerts",
```
Comment (TianoMS) [inline tests/2017-03-01-preview/Microsoft.Insights.ActionGroup.tests.json]: This should be "Microsoft.Insights/actionGroups". Same for the other tests in this file.

**Comment by TianoMS** [comment]
@dkamstra The test cases are failing. Please make sure they all pass.

File: schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json [hunk 6]
```diff
@@ -0,0 +1,196 @@
+{
+  "id": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.json#",
```
Comment (TianoMS) [inline schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json]: The "id" value is incorrect.

File: tests/2017-03-01-preview/Microsoft.Insights.ActionGroup.tests.json [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+              "countryCode": "1",
+              "phoneNumber": "5555551212"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "name": "Action Group Test - Email Only",
+      "definition": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.ActionGroup.json#/resourceDefinitions/actiongroups",
```
Comment (TianoMS) [inline tests/2017-03-01-preview/Microsoft.Insights.ActionGroup.tests.json]: Should be "actionGroups". Please fix for all test cases.

**Comment by dkamstra** [comment]
Just committed fixes for the latest feedback.

File: tests/2017-03-01-preview/Microsoft.Insights.ActivityLogAlerts.tests.json [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    {
+      "name": "Activity Log Alert - Service Health",
+      "definition": "http://schema.management.azure.com/schemas/2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json#/resourceDefinitions/activityLogAlerts",
+      "json": {
+        "type": "Microsoft.Insights/activityLogAlerts",
+        "apiVersion": "2017-03-01-preview",
+        "location": "Global",
+        "name": "TEST ServiceHealth Activity Log Alert",
+        "properties": {
+            "scopes": [
```
Comment (TianoMS) [inline tests/2017-03-01-preview/Microsoft.Insights.ActivityLogAlerts.tests.json]: Please format this file and use consistent indentation. We always use 2 spaces for indent.

File: tests/2017-03-01-preview/Microsoft.Insights.ActivityLogAlerts.tests.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+              "allOf": [
+                {
+                  "field": "category",
+                  "equals": "ServiceHealth"
+                }
+              ]
+            },
+            "actions": {
+                "actionGroups": [
+                {
```
Comment (TianoMS) [inline tests/2017-03-01-preview/Microsoft.Insights.ActivityLogAlerts.tests.json]: Indent.

**Pull request merged**